### ISSUE TITLE
Train- Test split unclear

### ISSUE THREAD

**Issue by Jostarndt** [kleelab-bch/FIGI-Net#4]
Dear authors,

Unfortunately I cannot run the code of 2_FIGINet_Prediction.py due to #2 . However I try to understand some training details and wanted to understand the train / test ( / validation ?) split.

In  the file `2_FIGINet_Prediction.py` the constant `Split_train = 75` in the code remains unused.

The block

 ```
df_New_cases_current_cluster_test_Extra = df_New_cases_diff_denoised_test[np.where(Label_idx[:] == (cluster + 1))]

df_New_cases_current_cluster_test = df_New_cases_diff_denoised_train[np.where(Label_idx[:] == (cluster + 1))]

df_New_cases_current_cluster_test = np.hstack((df_New_cases_current_cluster_test, df_New_cases_current_cluster_test_Extra))

```

seems to me, like the test set is a subset of the train set? Am I misunderstanding something?

Thanks for your help

**Comment by MurMikeS** [comment]
HI,

Thank you for your question! Let me clarify a couple of things:

1. You're right! The variable Split_train = 75 is not used in the current version of the code. It was left from an earlier version when I was experimenting with manual data splitting. You can safely ignore it.

2. Regarding the lines, this part might look confusing, but the intention is not to mix training and test sets for evaluation. This is actually a time series forecasting setup, where we need to provide initial input sequences from the past in order to generate predictions for the future.

        - df_New_cases_current_cluster_test (from the training set) is used as initial input data for the model.

        - df_New_cases_current_cluster_test_Extra (from the test set) contains the target prediction window.

These are concatenated because the model takes a continuous sequence to forecast the future daily values. However, during training and evaluation, the model is only trained on training data, and performance is evaluated only on the test part (i.e., future unseen steps).

So,  the test set **does not** include training data for evaluation. The concatenation is just to construct the full sequence needed to initialize the model and run the prediction.

Hope this clears things up!
